src/legacy/core_plugins/vis_type_table/public/table_vis_fn.ts

```diff
@@ -20,26 +20,53 @@
 import { i18n } from '@kbn/i18n';
 import { createTableVisResponseHandler } from './table_vis_request_handler';
 
-export const createTableVisFn  = (dependencies) => ({
-  name: 'kibana_table',
+import { ExpressionFunction, KibanaDatatable, Render } from '../../interpreter/types';
+
+const name = 'kibana_table';
+
+type Context = KibanaDatatable;
+
+interface Arguments {
+  visConfig: string | null;
+}
+
+type VisParams = Required<Arguments>;
+
+interface RenderValue {
+  visData: Context;
+  visType: 'table';
+  visConfig: VisParams;
+  params: {
+    listenOnChange: boolean;
+  };
+}
+
+type Return = Promise<Render<RenderValue>>;
+
+export const createTableVisFn = (): ExpressionFunction<
+  typeof name,
+  Context,
+  Arguments,
+  Return
+> => ({
+  name,
   type: 'render',
   context: {
-    types: [
-      'kibana_datatable'
-    ],
+    types: ['kibana_datatable'],
   },
   help: i18n.translate('visTypeTable.function.help', {
-    defaultMessage: 'Table visualization'
+    defaultMessage: 'Table visualization',
   }),
   args: {
     visConfig: {
       types: ['string', 'null'],
       default: '"{}"',
+      help: '',
     },
   },
   async fn(context, args) {
-    const visConfig = JSON.parse(args.visConfig);
-    const responseHandler = createTableVisResponseHandler(dependencies);
+    const visConfig = args.visConfig && JSON.parse(args.visConfig);
```

Comment:
nit: Is the added `args.visConfig &&` necessary here?

`JSON.parse(null)` returns `null`, so if `args.visConfig` is `null`, this check won't make a difference I think

Reply:
@lukeelmers 
![image](https://user-images.githubusercontent.com/20072247/62487952-c9006080-b7cb-11e9-9b24-84f789436873.png)

let's keep it as it

Reply:
@lukeelmers
![image](https://user-images.githubusercontent.com/20072247/62488133-390ee680-b7cc-11e9-8951-7be2ff8e894f.png)


let's keep it as it

Reply:
ahh makes sense. i didn't test this in TS so hadn't realized it would complain.